src/main/scala/DateFormat.scala

```diff
@@ -154,12 +154,29 @@ case class DateFormat(format: Option[String], dataType: String) {
     utcZoneId
   }
 
-  def getFormat() = format
+  def getFormat(): Option[String] = format
   val simpleDateFormatter: Option[SimpleDateFormat] = format.map(f => {
     ensureDateTimeFormatContainsRecognizedSymbols(f)
-    new SimpleDateFormat(f)
+    new SimpleDateFormat(mapTFormatsToUts35(f))
   })
 
+  private def mapTFormatsToUts35(formatIn: String): String = {
+    // As per W3C specification for CSV-W, datetime format patterns mentioned here in this method should also be accpected.
+    // The SimpleDateFormat class does not recognize the time separator(T)
+    // and thus these 3 formats are transformed to include the time separator (as a special char)
+    // https://www.w3.org/TR/2015/REC-tabular-data-model-20151217/#h-formats-for-dates-and-times
+    var format = formatIn
+    val replacement = Map(
+      "yyyy-MM-ddTHH:mm:ss.S" -> "yyyy-MM-dd'T'HH:mm:ss.S",
+      "yyyy-MM-ddTHH:mm:ss" -> "yyyy-MM-dd'T'HH:mm:ss",
+      "yyyy-MM-ddTHH:mm" -> "yyyy-MM-dd'T'HH:mm"
+    )
+    for ((original, newVersion) <- replacement) {
+      format = format.replaceAll(original, newVersion)
```

Comment:
Thinking about it again, there's only one replacement we actually need to do here. `      "yyyy-MM-ddTHH:mm" -> "yyyy-MM-dd'T'HH:mm"`

Let's remove this map and save ourselves some compute time.

Reply:
Yeah, true. I have changed it.